File: CPE/CPE49/32-UVA10190.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int m, n, k, tmp;
    bool flag;
    while (cin >> m >> n) {
        k = 0;
        flag = (m > 1 && n > 1)? true : false;
        if (flag){
            tmp = m;
            while (tmp % n == 0) {
                k++;
                tmp /= n;
            }
            if (tmp != 1) {
                flag = false;
            }
        }
        if (flag) {
            for (int i=k; i>=0; i--) {
                cout << m << (i ? " " : "\n");
                m /= n;
            }
        } else {
            cout << "Boring!" << endl;
        }
    }
    return 0;
}

File: CPE/CPE49/32-UVA10190.py
while True:
    try:
        m, n = map(int, input().split())
    except EOFError:
        break
    k = 0
    flag = True
    if m > 1 and n > 1:
        tmp = m
        while tmp % n == 0:
            k += 1
            tmp //= n
        if tmp != 1:
            flag = False
    else:
        flag = False
    if flag:
        for i in range(k, -1, -1):
            print(m, end=" " if i else "\n")
            m //= n
    else:
        print("Boring!")